src/extensions/posthog.ts

```diff
@@ -1,79 +1,81 @@
+            await producer.connect()
+            producer!.send({
+                topic: KAFKA_EVENTS_INGESTION_HANDOFF,
+                messages: [
+                    {
+                        key: data.uuid,
+                        value: JSON.stringify({
+                            distinct_id: data.distinct_id,
+                            ip: '',
+                            site_url: '',
+                            data: JSON.stringify(data),
+                            team_id: pluginConfig.team_id,
+                            now: data.timestamp,
+                            sent_at: data.timestamp,
+                            uuid: data.uuid,
+                        } as RawEventMessage),
+                    },
+                ],
+            })
         }
-
-        client!.sendTask(
-            'posthog.tasks.process_event.process_event_with_plugins',
-            [distinctId, null, null, data, pluginConfig.team_id, timestamp, timestamp],
-            {}
-        )
-    }
-
-    function sendEventKafka(event: string, properties: Record<string, any>, timestamp: string) {
-        const uuid = new UUIDT().toString()
-        const data = {
-            distinct_id: distinctId,
-            event,
-            timestamp,
-            properties: {
-                $lib: 'posthog-plugin-server',
-                $lib_version: version,
-                ...properties,
-            },
+    } else {
+        // Sending event to our Redis>Postgres pipeline
+        const client = new Client(server.redis, server.PLUGINS_CELERY_QUEUE)
+        sendEvent = async (data) => {
+            client.sendTask(
+                'posthog.tasks.process_event.process_event_with_plugins',
+                [data.distinct_id, null, null, data, pluginConfig.team_id, data.timestamp, data.timestamp],
+                {}
+            )
         }
-
-        producer!.send({
-            topic: KAFKA_EVENTS_WAL,
-            messages: [
-                {
-                    key: uuid,
-                    value: JSON.stringify({
-                        distinct_id: distinctId,
-                        ip: '',
-                        site_url: '',
-                        data: JSON.stringify(data),
-                        team_id: pluginConfig.team_id,
-                        now: timestamp,
-                        sent_at: timestamp,
-                        uuid,
-                    } as RawEventMessage),
-                },
-            ],
-        })
     }
 
-    const sendEvent = server.KAFKA_ENABLED ? sendEventKafka : sendEventRedis
-
     return {
         capture(event, properties = {}) {
-            const { timestamp, ...otherProperties } = properties
-            sendEvent(event, otherProperties, timestamp || DateTime.utc().toISO())
+            const { timestamp = DateTime.utc().toISO(), ...otherProperties } = properties
+            const data: InternalData = {
+                distinct_id: distinctId,
```

Comment:
Out of scope, but writing down just to write it down: I can see how spoofing "distinct_id" can be useful for plugins. For example when importing stuff from elsewhere or even just making `$set` queries to add data to users. 

It's a small change, but taking `distinct_id` from the data above (or `distinctId`?) like is done with the timestamp can unlock many future uses.